data/src/test/java/org/apache/iceberg/data/TestLocalScan.java

```diff
@@ -281,36 +271,33 @@ public void testFullScan() {
     expected.addAll(file3FirstSnapshotRecords);
 
     Set<Record> records = Sets.newHashSet(results);
-    Assert.assertEquals(
-        "Should produce correct number of records", expected.size(), records.size());
-    Assert.assertEquals("Random record set should match", Sets.newHashSet(expected), records);
+    Assertions.assertThat(records).as("Should produce correct number of records")
+        .hasSameSizeAs(expected);
+    Assertions.assertThat(records).as("Random record set should match")
+        .hasSameSizeAs(expected);
   }
 
-  @Test
+  @TestTemplate
   public void testFilter() {
     Iterable<Record> result = IcebergGenerics.read(sharedTable).where(lessThan("id", 3)).build();
 
-    Assert.assertEquals(
-        "Records should match file 1",
-        Sets.newHashSet(file1FirstSnapshotRecords),
-        Sets.newHashSet(result));
-
+    Assertions.assertThat(Sets.newHashSet(file1FirstSnapshotRecords)).as("Records should match file 1")
+        .isEqualTo(Sets.newHashSet(result));
```

Comment:
I believe actual/expected in the assertions here are the wrong way around and need to be fixed here in a few other places in this test